GitUI/MessageBoxes.cs

```diff
@@ -54,6 +57,12 @@ internal MessageBoxes()
 
         private static MessageBoxes Instance => instance ??= new();
 
+        public static void RevisionFilteredInGrid(IWin32Window? owner, ObjectId objectId)
+            => ShowError(owner, string.Format(Instance._cannotFindRevisionFilter.Text, objectId.ToShortString()), Instance._cannotFindRevisionCaption.Text);
```

Comment:
In a follow-up PR, we should switch `MessageBoxes` to `TaskDialog` and add an `Action` in order to turn off the filter directly.
Then this `Action` should retry to locate the revision.

Reply:
I do not think we can disable all possible filters. For instance all commits may not be loaded.
But it could be a suggestion, *Advanced filter* has most common occurrences (and could add more).

Reply:
`FilterInfo` contains all necessary information to know whether filters are applied or not. If there's something missing we need to add those checks there.
If there is no filter applied (as far as `FilterInfo` is concerned) we should show a task dialog with a message "Requested revision cannot be found". Else, if there is a filter applied, we should show a task dialog with an option to reset the filter - if that option is invoked - open the Advanced Filter dialog.

Reply:
> `FilterInfo` contains all necessary information to know whether filters are applied or not. If there's something missing we need to add those checks there. If there is no filter applied (as far as `FilterInfo` is concerned) we should show a task dialog with a message "Requested revision cannot be found". Else, if there is a filter applied, we should show a task dialog with an option to reset the filter - if that option is invoked - open the Advanced Filter dialog.

 `FilterInfo` is just a subset, there are several others like "--first-parent, --reflog, --simplify-by-decoration" at a quick glance.
Advanced filter will be quite bigger...

Only a few of the uses is from FormBrowse, so the AdvancedFilter is not accissible, an action to reset the filter could be used though.